coordinator/scripts/luigi_scoop_server.py

```diff
@@ -0,0 +1,206 @@
+    self.scooping_status_ = scooping_status
+    print(scooping_status)
+
+  # Arguments are of Pose type
+  def TranslationalPTPMotionClient(self, startPosition, endPosition):
+    rospy.wait_for_service('scooping_planning/scoop')
+    try:
+      # TODO: Only call this when status is IDLE
+      TranslationalPTPMotionServClient = rospy.ServiceProxy('scooping_planning/scoop', TranslationalPTPMotion)
+      resp = TranslationalPTPMotionServClient(startPosition, endPosition)
+      return resp.success
+    except rospy.ServiceException, e:
+      print "Service call failed: %s"%e
+
+  def SendFeedbackLuigi(self, feedback):
+    for i in scoops:
+      self._feedback.finished_scoops.append(0)
+    self._feedback.status_message = 'This is feedback'
+    # publish the feedback 
+    self.server_.publish_feedback(self._feedback)
+
+
+  # For now we set which flavor is where by hand (manually), unless vision can provide us with an api
+  def GoToFlavor(self, flavor):
+    leftStartPoint = Point(x=-0.1, y=-0.4, z=0.25)
+    rightStartPoint = Point(x=-0.1, y=-0.4, z=0.25)
+    startingPoint = Point()
+
+    # Left ice cream box
+    if flavor == availableFlavors[0]:
+      startingPoint = leftStartPoint
+    elif flavor == availableFlavors[1]:
+      startingPoint = rightStartPoint
+    # Do we fancy a third ice cream box?
+    else:
+      startingPoint = leftStartPoint
+
+    return startingPoint
+
+    
+  def RecieveIceCreamOrder_(self, data):
+
+    # success of the total scoping operation
+    success = False
+
+    # Scoops required
+    scoops = data.scoops
+
+    # Scoops done
+    scoopsScooped = np.zeros(len(scoops))
+
+    # flavors for each "scoops" ordered, not scoop :D
+    flavors = data.flavors
+
+    self.scooping_human_status_ = 'Recived a new ice cream order'
+    # intially no scoops has been scooped
+    for i in scoops:
+      self._feedback.finished_scoops.append(0)
+
+    scoops = data.scoops
+    flavors = data.flavors
+
+    r = rospy.Rate(1)
+
+    # start executing the action
+    # check that preempt has not been requested by the client
+    if self.server_.is_preempt_requested():
+      rospy.loginfo('%s: Preempted' % self._action_name)
+      self.server_.set_preempted()
+      success = False
+
+    # Go to the start point, Arne should give this point from simulation
+    startPoint = Point(x=-0.1, y=-0.4, z=0.25)
+
+    # TODO: Check the successs of the return
+    scoopingResponse = self.TranslationalPTPMotionClient(startPoint, startPoint)
+
+    # success of the total scoping operation
+    success = True
+    # We try to scoop in just five points sequence
+    for scoop in range(len(scoops)):
+
+      # Go to the start point of the box, depending on the flavor
+      self.GoToFlavor(flavors[scoop])
+
+      # Loop for how many scoops of this flavor, really luigi??
+      for scoopPerFlavor in range(scoops[scoop]):
+        
+        # Each scoop is done in scoopingSteps
```

Comment:
Well, yes, but this happens internally in the scooping module. You just call the scooping action once and the scooping side handles:
- move to the approach for scooping
- performing the scoop
- depart from the ice cream box
- dropping the ball

So I think on your side this is only a single call to the /scooping_planning/scoop service.

Reply:
Wait what :D?

Reply:
getFlavorPoint()